parsec/mca/pins/pins.c

```diff
@@ -31,6 +33,19 @@ void parsec_pins_instrument(struct parsec_execution_stream_s* es,
                             parsec_task_t* task)
 {
     assert( method_flag < PARSEC_PINS_FLAG_COUNT );
+    /**
```

Comment:
To discuss: it's nice to put this check here, and not in each cb_func, because this removes code duplication, but at the same time, the test below changes the behavior of PINS:

 - if PARSEC_PROF_TRACE is OFF, and task != NULL, no callback is called. This could be less intrusive by protecting the entire block (lines 40 to 47) behind the #ifdef
 - if a PINS module doesn't use parsec_profiling in cb_func, but uses it to take some other statistics, it's not called anyway.

I think we should pay the cost of code duplication here, to keep the feature that some pins module can collect statistics without profiling.

Reply:
w.r.t. line 897 in `jdf2c.c`, you're right. Oops, shouldn't have left that in. Fixed.

I'll be checking this email occasionally while off work. Just wanted to let you know I still see this stuff. I added a commit to change this, but I don't know what you mean by each callback function (`cb_func`). Where is this field set? Should we just write a wrapper that checks for the appropriate fields to be non-NULL and _then_ calls the cb_func?